Errors with v0.10.x when generating YAML

When I'm trying to run a build task with a buf.yaml file present, I get errors when it's trying to write the generated yaml file:

```
Caused by: java.lang.NoSuchMethodError: 'void org.yaml.snakeyaml.parser.ParserImpl.<init>(org.yaml.snakeyaml.reader.StreamReader, org.yaml.snakeyaml.LoaderOptions)'	
at com.fasterxml.jackson.dataformat.yaml.YAMLParser.<init>(YAMLParser.java:196)	
at com.fasterxml.jackson.dataformat.yaml.YAMLFactory._createParser(YAMLFactory.java:505)	
at com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:393)	
at com.fasterxml.jackson.dataformat.yaml.YAMLFactory.createParser(YAMLFactory.java:15)	
at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3825)	
at build.buf.gradle.BufYamlGenerator.generate(BufYamlGenerator.kt:48)	
at build.buf.gradle.WriteWorkspaceYamlTask.writeWorkspaceYaml(ProtobufGradlePluginSupport.kt:147)
```

Dependency looks to be pulling in snakeyaml v2.4.

If I _don't_ specify a buf.yaml file and just run with defaults, everything works.  However, my use case needs the customization of buf.yaml.  Let me know if you have any suggestions or need more details!

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -80,7 +80,7 @@ ext[GRADLE_PUBLISH_SECRET] = System.getenv("GRADLE_PORTAL_PUBLISH_SECRET")
 
 kotlin {
     jvmToolchain {
-        languageVersion.set(JavaLanguageVersion.of(11))
+        languageVersion.set(JavaLanguageVersion.of(17))
     }
 }
 

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -9,7 +9,7 @@ buildConfig = "5.6.8"
 
 # runtime
 bufbuild = "1.59.0"
-jackson = "2.20.0"
+jackson = "3.0.1"
 protoc = "4.33.0"
 versioncompare = "1.5.0"
 
@@ -31,8 +31,8 @@ spotless = { id = "com.diffplug.spotless", version.ref = "spotless" }
 [libraries]
 # runtime
 bufbuild = { module = "build.buf:buf", version.ref = "bufbuild" }
-jacksonDataformatYaml = { module = "com.fasterxml.jackson.dataformat:jackson-dataformat-yaml", version.ref = "jackson" }
-jacksonModuleKotlin = { module = "com.fasterxml.jackson.module:jackson-module-kotlin", version.ref = "jackson" }
+jacksonDataformatYaml = { module = "tools.jackson.dataformat:jackson-dataformat-yaml", version.ref = "jackson" }
+jacksonModuleKotlin = { module = "tools.jackson.module:jackson-module-kotlin", version.ref = "jackson" }
 protoc = { module = "com.google.protobuf:protoc", version.ref = "protoc" }
 versioncompare = { module = "io.github.g00fy2:versioncompare", version.ref = "versioncompare" }
 

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.14.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.14.3-bin.zip
 networkTimeout=10000
 validateDistributionUrl=true
 zipStoreBase=GRADLE_USER_HOME

--- a/src/main/kotlin/build/buf/gradle/BufYamlGenerator.kt
+++ b/src/main/kotlin/build/buf/gradle/BufYamlGenerator.kt
@@ -1,13 +1,17 @@
 package build.buf.gradle
 
-import com.fasterxml.jackson.databind.ObjectMapper
-import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
-import com.fasterxml.jackson.module.kotlin.readValue
-import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import tools.jackson.dataformat.yaml.YAMLFactory
+import tools.jackson.dataformat.yaml.YAMLMapper
+import tools.jackson.module.kotlin.KotlinModule
+import tools.jackson.module.kotlin.readValue
 import java.io.File
 
 internal class BufYamlGenerator {
-    private val mapper = ObjectMapper(YAMLFactory()).registerKotlinModule()
+    private val mapper =
+        YAMLMapper
+            .builder(YAMLFactory.builder().build())
+            .addModule(KotlinModule.Builder().build())
+            .build()
 
     /**
      * Read the user-supplied buf.yaml file and generate an equivalent file, adding a

